immer/detail/rbts/rrbtree.hpp

```diff
@@ -49,13 +49,18 @@ struct rrbtree
 
     static node_t* empty_root()
     {
-        static const auto empty_ = node_t::make_inner_n(0u);
+        constexpr auto size = node_t::sizeof_inner_n(0);
+        static std::aligned_storage_t<size, alignof(std::max_align_t)> storage;
+        static const auto empty_ =
+            node_t::make_inner_n_into(&storage, size, 0u);
         return empty_->inc();
     }
 
     static node_t* empty_tail()
     {
-        static const auto empty_ = node_t::make_leaf_n(0u);
+        constexpr auto size = node_t::sizeof_leaf_n(0);
+        static std::aligned_storage_t<size, alignof(std::max_align_t)> storage;
+        static const auto empty_ = node_t::make_leaf_n_into(&storage, size, 0u);
```

Comment:
I think I would prefer this:
```c++
static const auto empty_ = [] {
  static std::aligned_storage_t<size, alignof(std::max_align_t)> storage;
  return node_t::make_leaf_n_into(&storage, size, 0u);
}();
```
This hides `storage` inside the initialization of `empty_` which I think could have some advantages (from an understandability POV, but it also makes it clear to the compiler that it only needs to access it once, during the protected initialization of `empty_` itself.)

Reply:
The same self-executing lambda pattern can be used for the root above.

Reply:
Very good!